### PR TITLE
Check for expected changes after CDI upgrade

### PR DESCRIPTION
**Pull request by arnongilboa**
**What this PR does / why we need it**:

Upgrade testing should verify all cdi pod names, uids and image versions changed (CNV-5252).

**Special notes for your reviewer**:

**Release note**:
```NONE
```

### CODE REVIEW

**Comment by arnongilboa** [comment]
It looks like the patch is a bit buggy. Working on a fix.

**Comment by arnongilboa** [comment]
@awels can you please review?

File: cluster-sync/sync.sh [hunk 1]
```diff
@@ -79,6 +80,40 @@ function wait_cdi_available {
   fi
 }
 
+OLD_CDI_VER_PODS="/tmp/old_cdi_ver_pods"
```
Comment (mhenriks) [inline cluster-sync/sync.sh:83]: maybe `./_out/tests` is better than `/tmp`?

Comment (arnongilboa) [reply cluster-sync/sync.sh:83]: sure

File: cluster-sync/sync.sh [hunk 2]
```diff
@@ -79,6 +80,40 @@ function wait_cdi_available {
...
+  if [ -f $NEW_CDI_VER_PODS ] ; then
+    mv $NEW_CDI_VER_PODS $OLD_CDI_VER_PODS
+  fi
+  wait_time=0
+  ret=0
+  while [[ $ret -eq 0 ]] && [[ $wait_time -lt ${CDI_PODS_UPDATE_TIMEOUT} ]]; do
+    wait_time=$((wait_time + 5))
+    _kubectl get pods -n $CDI_NAMESPACE -o=jsonpath='{range .items[*]}{.metadata.name}{"\n"}{.metadata.uid}{"\n"}{.spec.containers[*].image}{"\n"}{end}' > $NEW_CDI_VER_PODS
+    if [ -f $OLD_CDI_VER_PODS ] ; then
+      grep -qFxf $OLD_CDI_VER_PODS $NEW_CDI_VER_PODS || ret=$?
```
Comment (mhenriks) [inline cluster-sync/sync.sh:99]: Will this return non zero when there is any difference between old and new?  If so, is that really sufficient to say that all pods have changed?

Comment (arnongilboa) [reply cluster-sync/sync.sh:99]: @mhenriks no, grep will return non-zero when **none** of the lines (pod name, uid or image) on the old file appears in the new file, which sure means all pods have changed. However it does not check all the new pods are Running etc., which out of scope here.

**Pull request merged**